adr/0006-native-compilation-with-binary-libraries.adoc

```diff
@@ -0,0 +1,155 @@
+= Including Native Libraries in the Native Image
+
+* Status: _Proposed_
+* Date: 2024-10-21
+* Authors: @cescoffier, @zakkak
+
+== Context and Problem Statement
+
+At Quarkus’ scale, several extensions integrate libraries that require native code.
+Most of these libraries rely on JNI (Java Native Interface) to interact with native code.
```

Comment:
how about those that does not use JNI? JNA? external binaries?

Reply:
For most of them using build steps in the _deployment_ module should be the way to go.

Reply:
Would be good to spell that out as we know of at least  a few extension relying on non-jni (although very much the exception)

Reply:
I've added a section.